mbuild/lib/recipes/water_box.py

```diff
@@ -0,0 +1,231 @@
+                    "The first particle in model needs to correspond to oxygen."
+                )
+            if len(particles) != 3:
+                raise MBuildError("The only works with 3-site models of water.")
+
+        # check if mask is set
+        if mask is not None:
+            if not isinstance(mask, list):
+                if not isinstance(mask, Compound):
+                    raise MBuildError(
+                        f"Mask must be a Compound or a list of Compounds."
+                    )
+
+            elif isinstance(mask, list):
+                # in case we are specified a list of Compounds,
+                # we will make sure it is a 1d list.
+                mask = [e for e in _flatten_list(mask)]
+                for entry in mask:
+                    if not isinstance(entry, Compound):
+                        raise MBuildError(
+                            f"Mask must be a Compound or a list of Compounds."
+                        )
+
+        # read in our propotype, a 4.0x4.0x4.0 nm box
+        # our prototype was relaxed in GROMACs at 305 K, density 1000 kg/m^3 using tip3p
+        aa_waters = load(
+            "water_proto.gro",
+            relative_to_module=self.__module__,
+        )
+
+        # loop over each water in our configuration
+        # add in the necessary bonds missing from the .gro file
+        # rename particles/Compound according to the given water model
+        for water in aa_waters.children:
+            water.add_bond((water.children[0], water.children[1]))
+            water.add_bond((water.children[0], water.children[2]))
+
+            temp = clone(model)
+            force_overlap(temp, temp, water, add_bond=False)
+            water.name = model.name
+            water.children[0].name = model.children[0].name
+            water.children[1].name = model.children[1].name
+            water.children[2].name = model.children[2].name
+            water.xyz = temp.xyz
+
+        # scaling parameters for the new box
+        scale_Lx = math.ceil(box.Lx / aa_waters.box.Lx)
+        scale_Ly = math.ceil(box.Ly / aa_waters.box.Ly)
+        scale_Lz = math.ceil(box.Lz / aa_waters.box.Lz)
+
+        water_system_list = []
+
+        # we will create a list of particles for the mask
+        # if specified now to save time later
+        if mask is not None:
+            if isinstance(mask, Compound):
+                p_mask = [p for p in mask.particles()]
+            else:
+                p_mask = []
+                for entry in mask:
+                    p_mask = p_mask + [p for p in entry.particles()]
+
+        # add water molecules to a list
+        # note we add to a list first, as this is more efficient than calling
+        # the Compound.add function repeatedly as the Compound size grows.
+        for water in aa_waters.children:
+            for i in range(0, scale_Lx):
+                for j in range(0, scale_Ly):
+                    for k in range(0, scale_Lz):
```

Comment:
```suggestion
            for i, j, k in itertools.product(range(scale_Lx), range(scale_Ly), range(scale_Lz)):
```
may need to import `itertools` if it's not yet imported.